[doc] First draft for demo pages

MaxPal opened the pull request:
Documentation: First draft for demo page.

MaxPal committed:
[doc] Example of page for demo case

MaxPal committed:
{doc]  Pythoin script to read h5

Extract data for documentation.

MaxPal committed:
[doc] Demo page. Use python to plot graph

MaxPal committed:
ignore  python files

MaxPal committed:
[doc] First draft for demo pages

MaxPal commented:
![Capture d’écran du 2023-06-30 17-51-52](https://github.com/NREL/HydroChrono/assets/9744087/81abd221-1435-4ef2-835d-b0751c6d2aba)

dav-og committed:
ask dev to explicitly define their conda environment being used (with h5py, sphinx etc)

dav-og committed:
comment out hard coded path

dav-og committed:
new docs structure and 'homepage'

dav-og committed:
added to 'getting started' section

MaxPal committed:
Find python and sphinx. fix for Linux

MaxPal committed:
[cmake] Helper to findPython3 or Sphinx

MaxPal committed:
Merge branch 'main' into mpa/doc3

MaxPal commented:
@dav-og I reversed the detection of python3 and sphinx to the original. I have just added some messages with hints.

If detection of python3 fails you can give hints in the CMakeCache.txt file as mentioned in the cmake documentation at 

https://cmake.org/cmake/help/latest/module/FindPython3.html

You can provide the path to the interpreter's directory with
> Python3_ROOT_DIR=...  

or directly give 

> Python3_EXECUTABLE=...

and for virtual python environment you can give a strategy with 
> Python3_FIND_VIRTUALENV=...

according to this one, conda environment are also manged since 3.17 version of cmake

dav-og committed:
include option to define python environment...if path is provided, use python and sphinx from there

dav-og commented:
Hi @MaxPal, I couldn't compile the docs with this approach as it wasn't calling the correct python.exe from my environment with sphinx etc...I tried modifying CMakeCache.txt but still couldn't get it building correctly and this file gets overwritten when the project is re-generated, which is a pain.

Hence, I included an optional variable in the cmake-gui for people to specify a python environment. If this option is taken, then cmake will find the python and sphinx executables from that environment. This is working for me now.

dav-og committed:
added theory docs

MaxPal commented:
@dav-og For citation in documentation we can use the Bibtex format but for this we need a sphinx extension ('sphinxcontrib.bibtex') to be installed in python environment (like for mathjax). 

https://chiplicity.readthedocs.io/en/latest/Using_Sphinx/UsingBibTeXCitationsInSphinx.html#id6

What do you think ?

dav-og commented:
> @dav-og For citation in documentation we can use the Bibtex format but for this we need a sphinx extension ('sphinxcontrib.bibtex') to be installed in python environment (like for mathjax).
> 
> https://chiplicity.readthedocs.io/en/latest/Using_Sphinx/UsingBibTeXCitationsInSphinx.html#id6
> 
> What do you think ?

Sounds good Maxime! 👍

On a different note, are the if statements I included [here](https://github.com/NREL/HydroChrono/pull/36/commits/64d9be72b373088a8f293838883a177b82e9bb0e) ok by you?

MaxPal committed:
[fix] detection of sphinx used an obsolete find python

FindPythonInterp package. Replaced by the standard FindPython3.

MaxPal committed:
[doc] todolist is made to be used with ..todo directive

reorganize todo in a specific page.

MaxPal committed:
[doc] cleaing. fix dome duplicates and bad references

MaxPal commented:
@dav-og The modification with PYTHON_ENV is not good for us because it forces some specific python bypassing the detection by cmake. If you realy want to by pass cmake you should directly modify the CMakeCache.txt file.

Anyway, I  think I fixed the problem in the commit https://github.com/NREL/HydroChrono/pull/36/commits/716d85490549ddae8c10780245d37f5b164ee5cc

The detection of sphinx used an obsolete cmake find package for python, different of the one I used for python3. If you want, we can organize a quick visio so you will be able to share your screen in order to help you configure the good environment with conda

MaxPal committed:
[doc] bibtex in sphinx. 1st try

MaxPal committed:
[doc] delete duplicates

MaxPal committed:
[doc] fix indentation

MaxPal committed:
[doc] fix bad path for python script + missing footnote

MaxPal committed:
[doc] add a specific page for bibliography

MaxPal committed:
[doc] reverse Python_env hint

dav-og committed:
cmake fix to to find sphinx near to python

changed PYTHONINTERP_FOUND to Python3_FOUND

added 'STATUS' to cmake messages

MaxPal committed:
Use same find python3 dor documlentation and tests

replace hard coded python3 by variable ${Python3_EXECUTABLE} for test commands

dav-og committed:
included additional instructions to create 'Python3_ROOT_DIR' variable for cmake and point it to environment

python dependencies:

- matplotlib
- sphinx
- sphinxcontrib-bibtex
- h5py

consider including a .yml file to ensure correct versions and compatibility across systems.

dav-og committed:
cleaned up sphere verification section

dav-og committed:
some src docs with Doxygen

need to install breathe in python environment too

src needs work to auto document properly

dav-og committed:
small fix to src docs and new section for oswec

dav-og committed:
filled in oswec verification section

dav-og committed:
resize oswec images

dav-og committed:
fix some doxygen things

dav-og committed:
change all .cpp paths to .h for doxygen

dav-og committed:
included about section

dav-og committed:
Update about.rst

MaxPal commented:
@dav-og With the addition of the Doxygen documentation (doxygen + breathe dependencies) for sphinx documentation the compilation is broken (on my linux workstation).  I propose in the first place to set up  an option to deactivate the doxygen part. So we will be able to quickly merge this branch. In a second time we should generalized the use of Doxygen but it takes more time to put in place.  What do you think ?

MaxPal committed:
[doc] [fix] Doxygen optional for sphinx

Detection of sphinx python modules:
 - Found python 'sphinxcontrib.bibtex'
 - Found python 'breathe'

 if breateh is not found it is not set in the configuration conf.py.in of sphinx

 You can disable usage of doxygen be setting general option
 HYDROCHRONO_ENABLE_PROG_DOC to false

 WARNING: The generated index.xml file of doxygen is still hard coded in conf.py.in of sphinx !

MaxPal commented:
@dav-og Can you check my last commit https://github.com/NREL/HydroChrono/pull/36/commits/56abae8e96494e40ee36353068ba4eb9dc0a1100 is fine for you.

Doxygen optional for sphinx

Detection of sphinx python modules:

 - python 'sphinxcontrib.bibtex'
 - python 'breathe'

 if breateh is not found it is not set in the configuration conf.py.in of sphinx

 You can disable usage of doxygen be setting general option
 
> HYDROCHRONO_ENABLE_PROG_DOC=FALSE

 WARNING: The generated index.xml file of doxygen is still hard coded in conf.py.in of sphinx !

MaxPal committed:
[doc] deactivate doxygen
